python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = False
+
+
+class DefectMockLSST(IsrMockLSST):
+    """Simulated defect list.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+        self.config.doBrighterFatter = False
+        self.config.doDefects = True
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = False
+
+
+class CrosstalkCoeffMockLSST(IsrMockLSST):
+    """Simulated crosstalk coefficient matrix.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+        self.config.doBrighterFatter = False
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = True
+        self.config.doTransmissionCurve = False
+
+
+class TransmissionMockLSST(IsrMockLSST):
+    """Simulated transmission curve.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+        self.config.doBrighterFatter = False
+        self.config.doDefects = False
+        self.config.doCrosstalkCoeffs = False
+        self.config.doTransmissionCurve = True
+
+
+class MockLSSTDataContainer(object):
+    """Container for holding ISR LSST mock objects.
+    """
+    dataId = "isrMockLSST Fake Data"
+    darkval = 2.  # e-/sec
+    oscan = 250.  # DN
+    gradient = .10
+    exptime = 15.0  # seconds
+    darkexptime = 15.0  # seconds
+
+    def __init__(self, **kwargs):
+        if 'config' in kwargs.keys():
+            self.config = kwargs['config']
+        else:
+            self.config = None
+
+    def expectImage(self):
+        if self.config is None:
+            self.config = IsrMockLSSTConfig()
+        self.config.doGenerateImage = True
+        self.config.doGenerateData = False
+
+    def expectData(self):
+        if self.config is None:
+            self.config = IsrMockLSSTConfig()
+        self.config.doGenerateImage = False
+        self.config.doGenerateData = True
+
+    def get(self, dataType, **kwargs):
+        """Return an appropriate data product.
+
+        Parameters
+        ----------
+        dataType : `str`
+            Type of data product to return.
+
+        Returns
+        -------
+        mock : IsrMockLSST.run() result
+            The output product.
+        """
+        if "_filename" in dataType:
+            self.expectData()
+            return tempfile.mktemp(), "mock"
+        elif 'transmission_' in dataType:
+            self.expectData()
+            return TransmissionMockLSST(config=self.config).run()
+        elif dataType == 'ccdExposureId':
+            self.expectData()
+            return 20090913
```

Comment:
Can we put a date in the middle of the survey?  (It doesn't matter but would be nice)

Reply:
(And for all the dates in this file)

Reply:
I am removing the containers as they are out of date and I don't see their use at the moment.